src/sentry/notifications/platform/service.py

```diff
@@ -233,3 +239,39 @@ def notify_target_async[T: NotificationData](
             lifecycle.record_halt(halt_reason=e, create_issue=False)
         except Exception as e:
             lifecycle.record_failure(failure_reason=e, create_issue=True)
+
+
+@dataclass
+class NotificationDataDto:
+    """
+    A wrapper class that handles serialization/deserialization of NotificationData.
+    """
+
+    notification_data: NotificationData
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "source": self.notification_data.source,
+            "data": self.notification_data.__dict__,
+        }
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "NotificationDataDto":
+        source = data.get("source")
+        if source is None:
+            raise NotificationServiceError("Source is required")
+
+        notification_data_cls = template_registry.get(source).get_data_class()
+        notification_fields = data.get("data", {}).copy()
+
+        # We're using type hints to know which fields need special conversion
+        type_hints = get_type_hints(notification_data_cls)
+
+        for field_name, value in notification_fields.items():
```

Comment:
All of this serialization, deserialization has me wondering if we should just convert to using pydantic instead.